src/main/groovy/edu/oregonstate/mist/cardsapi/resources/CardsResource.groovy

```diff
@@ -168,6 +168,21 @@ class CardsResource extends Resource {
                     "must match pattern: " + validPattern).build()
         }
 
+        // Validate energy
+        if(energyMin.present && energyMax.present &&
+                !(energyMin.get() >= 0 && energyMin.get() <= 999 &&
+                        energyMax.get() >= 0 && energyMax.get() <= 999)) {
+            return badRequest("energyMin or energyMax out of range. " +
+                    "Valid range is 0 to 999.").build()
+        }
+        if(energyMin.present ^ energyMax.present) {
+            return badRequest("energyMin and energyMax must both be valid " +
+                    "or must both be null.").build()
+        }
+        if(energyMin.present && energyMax.present && !(energyMin.get() <= energyMax.get())) {
+            return badRequest("energyMin must not be greater than energyMax.").build()
+        }
+
```

Comment:
Could you use variables to save these values since you called them multiple times?